src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                }
+
+                // Consider a mutation request such as the following
+                // mutation{
+                //  createbooks(items: [
+                //                {
+                //                    title: "Harry Potter and the Chamber of Secrets",
+                //                    publishers: { name: "Bloomsbury" }
+                //                 },
+                //                 {
+                //                    title: "Educated",
+                //                    publishers: { name: "Random House"}    
+                //                 }
+                //     ]){
+                //      items{
+                //         id
+                //         title 
+                //         publisher_id 
+                //      }
+                //   }
+                // }
+                // For the above mutation request, in the parsedInputList, the 0th dictionary will correspond to the fields for the 0th element in the items array.  
+                // Likewise, 1st dictionary in the parsedInputList will correspond to the fields for the 1st element in the items array and so on.
+                // Each element in the items array is independent of any other element in the array. Therefore, the create operation for each element in the items array is independent of the other elements.
+                // So, parsedInputList is iterated and the create operation is performed for each element in the list.
+                foreach (IDictionary<string, object?> parsedInput in parsedInputList)
+                {
+                    MultipleCreateStructure multipleCreateStructure = new(entityName: entityName,
+                                                                          higherLevelEntityName: entityName,
+                                                                          higherLevelEntityPKs: null,
+                                                                          inputMutParams: parsedInput);
+
+                    Dictionary<string, Dictionary<string, object?>> primaryKeysOfCreatedItem = new();
+
+                    IValueNode? fieldNodeForCurrentItem = paramList[idx];
+                    if (fieldNodeForCurrentItem is null)
+                    {
+                        throw new DataApiBuilderException(message: "Error when processing the mutation request",
+                                                          statusCode: HttpStatusCode.InternalServerError,
+                                                          subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+                    }
+
+                    PerformDbInsertOperation(context, fieldNodeForCurrentItem.Value, sqlMetadataProvider, multipleCreateStructure, primaryKeysOfCreatedItem);
+                    if (multipleCreateStructure.CurrentEntityPKs is not null)
```

Comment:
True. `CurrentEntityPKs` will ideally not be `null`. If it were to be `null`, some problem had occurred when trying to create a record in the database at which point an exception would get thrown. 

The `null` check is just defensive code to prevent having to deal with `null` values in the selection set resolution.